schema/crdb/dbinit.sql

```diff
@@ -238,7 +238,13 @@ CREATE TABLE IF NOT EXISTS omicron.public.sled_resource (
     reservoir_ram INT8 NOT NULL,
 
     -- Identifies the type of the resource
-    kind omicron.public.sled_resource_kind NOT NULL
+    kind omicron.public.sled_resource_kind NOT NULL,
+
+    -- The UUID of an instance, if this resource belongs to an instance.
+    instance_id UUID
```

Comment:
... I do, however, want to know the instance UUID of this object, if it exists. 

I don't *think* I can rely on the `omicron.public.vmm` table existing by the time reservations are made - note that e.g. for instance start, "allocating a server" happens before "VMM record creation":

https://github.com/oxidecomputer/omicron/blob/7f05fce624a14fdbeea989e7e959617d626a827d/nexus/src/app/sagas/instance_start.rs#L136-L138

So - when we create a sled_resource record on behalf of an instance, we need a way for *other* requests to identify "what instance does this VMM belong to?". Hence the addition of this optional field.